test/utils/driver.js

```diff
@@ -72,6 +72,40 @@ const Driver = {
     this.app = this.port = this.fixturesURL = this._stopServer = this._bwHandle = null;
   },
 
+  /**
+   * Fetches the redux state of the first browser window and returns
+   * it as a plain JS object (de-immutable-ified).
+   *
+   * @return Promise<Object>
+   */
+  getReduxState: async function() {
+    await this.setTargetToBrowserWindow();
+
+    return await this.client.execute(() =>
+      JSON.stringify(require('electron').ipcRenderer.store.getState()))
+      .then(res => JSON.parse(res.value));
+  },
+
+  /**
+   * Polls the redux state of the first browser window
+   * until the passed in predicate, which receives the state as a
+   * plain JS Object, returns a truthy value, or resolves to a truthy value.
+   *
+   * @param {Function} predicate
+   * @return Promise
+   */
+  waitUntilReduxState: async function(predicate) {
```

Comment:
I'm fine with this approach.  I'm not a huge fan of polling in this way, but c'est la vie.